### DISCUSSION TITLE
problem fetching data using post method returns bad request :(

### DISCUSSION THREAD

**Discussion by heleopless50** [serenity-is/Serenity#6786] [General]
i am trying to fetch data using post method but it returns bad request 
but when i use fetch data using get it works fine 
i dont know the problem - i tried to debug but it doesnt reach the endpoint to execute code in it ! please help me ! 
this is my code  :
          fetch('/Accounts/Accounts/RenameAccount', {
            method: 'POST',
            headers: {
              'Content-Type': 'application/json'
            },
            body: JSON.stringify({ name: "ahmed", city: "hehia" })
          }).then(data => console.log(data));
          
        ///////////////////////
        and this is the end point code 
        [HttpPost]
[Route("Accounts/Accounts/RenameAccount")]
public JsonResult RenameAccount()
{
    using (StreamReader reader = new StreamReader(Request.Body))
    {
        string requestBody = reader.ReadToEnd();
        var data = JsonConvert.DeserializeObject<dynamic>(requestBody);

        // Access the values from the deserialized data
        var accountId = (int)data.accountId;
        var newName = (string)data.newName;

        // Handle the data and perform necessary operations

        return new JsonResult(true);
    }
}

**Comment by wezmag** [comment]
I think it is due to the antiforgery token was not submitted with the request.
You can test it by adding `[IgnoreAntiforgeryToken]` attribute to your end point.
If it works, you need to find a way to include antiforgery token to your request.

**Reply by heleopless50** [reply]
that is great , it worked 
**thank you so much :)**
